cmd/kube-scheduler/app/server.go

```diff
@@ -155,86 +145,86 @@ func runCommand(cmd *cobra.Command, args []string, opts *options.Options) error
 
 	// Configz registration.
 	if cz, err := configz.New("componentconfig"); err == nil {
-		cz.Set(cc.ComponentConfig)
+		cz.Set(opts)
 	} else {
 		return fmt.Errorf("unable to register configz: %s", err)
 	}
 
-	return Run(cc, stopCh)
+	return Run(opts, stopCh)
 }
 
 // Run executes the scheduler based on the given configuration. It only return on error or when stopCh is closed.
-func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}) error {
+func Run(opts *options.Options, stopCh <-chan struct{}) error {
 	var storageClassInformer storageinformers.StorageClassInformer
 	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeScheduling) {
-		storageClassInformer = cc.InformerFactory.Storage().V1().StorageClasses()
+		storageClassInformer = opts.InformerFactory.Storage().V1().StorageClasses()
 	}
 
 	// Create the scheduler.
-	sched, err := scheduler.New(cc.Client,
-		cc.InformerFactory.Core().V1().Nodes(),
-		cc.PodInformer,
-		cc.InformerFactory.Core().V1().PersistentVolumes(),
-		cc.InformerFactory.Core().V1().PersistentVolumeClaims(),
-		cc.InformerFactory.Core().V1().ReplicationControllers(),
-		cc.InformerFactory.Apps().V1().ReplicaSets(),
-		cc.InformerFactory.Apps().V1().StatefulSets(),
-		cc.InformerFactory.Core().V1().Services(),
-		cc.InformerFactory.Policy().V1beta1().PodDisruptionBudgets(),
+	sched, err := scheduler.New(opts.Client,
+		opts.InformerFactory.Core().V1().Nodes(),
```

Comment:
It seems to me that the fields which were moved from `Config` into `Options` are written once in `initialize()` and read once right here. We could probably have them as local variables in this function, rather than keeping them in the `Options` struct. What do you think?

Reply:
I move the old `Config` fields to `app.Run` as local variables, except `SecureServingInfo`, `InsecureServingInfo` and `InsecureMetricsServingInfo`, the reasons are as follows:

1. [testserver](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/testing/testserver.go) may use `LoopbackClientConfig` to create a k8s client before `app.Run`: https://github.com/kubernetes/kubernetes/blob/a78b591ed9fc42cb3f896d6a7f8178008dbf1239/cmd/kube-scheduler/app/testing/testserver.go#L129

that means we need to initialize `LoopbackClientConfig` inside `Options.Initialize`. Unfortunately, `InsecureServingInfo` and `InsecureMetricsServingInfo` are initialized along with `LoopbackClientConfig`: https://github.com/kubernetes/kubernetes/blob/a78b591ed9fc42cb3f896d6a7f8178008dbf1239/cmd/kube-scheduler/app/options/insecure_serving.go#L63

So change `InsecureServingInfo` and `InsecureMetricsServingInfo` as local variables makes initialization difficult to perform.

2. The integration test wants to verify `SecureServingInfo` and `InsecureServingInfo`, this requires these fields be exported: https://github.com/kubernetes/kubernetes/blob/a78b591ed9fc42cb3f896d6a7f8178008dbf1239/test/integration/serving/serving_test.go#L76

Reply:
That seems reasonable to me.

PS: How did you get the code snippets to show up in your comment?

Reply:
I just paste the code link in the comment. It seems that GitHub only show the code snippets for links of current repo(kubernetes). Code links of other repos(such as apiserver, apimachinery, etc.) won't be showed as snippets.

Reply:
@misterikkit try use "permanent link".

![image](https://user-images.githubusercontent.com/1425903/49920417-03914380-fe5f-11e8-83b2-a1ef6b316bc5.png)